File: frontend/src/interfaces/OwnerGame.ts
import type { GameUser } from "./GameUser"; 
import type { Game } from "./Game";

export interface OwnerGame {
  ID: number;
  PurchaseDate?: string;
  Status?: string;
  GameUserID?: number;
  GameID?: number;
  GameUser?: GameUser;
  Game?: Game;
}

File: backend/entity/workshop/ownergame.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type OwnerGame struct {
	gorm.Model
	
	PurchaseDate  time.Time
	Status  string

	GameUserID *uint
	GameUser   GameUser `gorm:"foreignKey:GameUserID"`

	GameID *uint
	Game   Game `gorm:"foreignKey:GameID"`
	
}